samples/api/hpp_hello_triangle/hpp_hello_triangle.cpp

```diff
@@ -307,10 +307,22 @@ vk::Device HPPHelloTriangle::create_device(const std::vector<const char *> &requ
 		throw std::runtime_error("Required device extensions are missing, will try without.");
 	}
 
+	std::vector<const char *> active_device_extensions(required_device_extensions);
+
+#if (defined(VKB_ENABLE_PORTABILITY))
+	// VK_KHR_portability_subset must be enabled if present in the implementation (e.g on macOS/iOS with beta extensions enabled)
+	if (std::any_of(device_extensions.begin(),
+	                device_extensions.end(),
+	                [](vk::ExtensionProperties const &extension) { return strcmp(extension.extensionName, VK_KHR_PORTABILITY_SUBSET_EXTENSION_NAME) == 0; }))
+	{
+		active_device_extensions.push_back(VK_KHR_PORTABILITY_SUBSET_EXTENSION_NAME);
+	}
+#endif
```

Comment:
Instead of copying over the complete vector `required_device_extensions` (ok, it has just one element...) I would prefer to have that extension be added to the `required_device_extension` where `create_device` is called (line 160).

Reply:
Thanks @asuessenbach.  I actually proptyped that option first since `required_device_extensions` is a `const`.  However I came back to creating and using `active_device_extensions` for a few reasons:

1. The spec requires you to check if `VK_KHR_portability_subset` is available before enabling, and thus needs `device_extensions` to be available, which it is not in the calling function.
2. `create_instance()` uses the same code pattern by copying `required_instance_extensions` into `active_instance_extensions` before adding additional extensions to the vector.
3. I thought it was better to encapsulate this logic in `create_device()` vs "pollute" the calling function.

Reply:
> create_instance() uses the same code pattern by copying required_instance_extensions into active_instance_extensions before adding additional extensions to the vector.

I see... it's been a while since I had a look at those samples.
That is: agreed, doing it this way is consistent with the rest of the code here.

Thanks for the careful editing.